File: Data Structures & Algorithms/reverse-a-linked-list/submission-1.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func reverseList(head *ListNode) *ListNode {
    if head == nil {
        return nil
    }
    var ans *ListNode

    for head != nil {
        next := head.Next
        head.Next = ans
        ans = head
        head = next
    }
    return ans
}

File: Data Structures & Algorithms/remove-node-from-end-of-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode(0, head)
        lp = dummy
        rp = head
        for _ in range(n - 1):
            rp = rp.next
        
        while rp.next is not None:
            lp = lp.next
            rp = rp.next

        lp.next = lp.next.next
        return dummy.next
        